core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/queue/SchedulingDecisionMaker.scala

```diff
@@ -79,7 +89,7 @@ class SchedulingDecisionMaker(
            *
            * However, if the container exists(totalContainers != 0), the activation is not treated as a failure and the activation is delivered to the container.
            */
-          case Running =>
+          case Running if totalContainers == 0 || !schedulingConfig.allowOverProvisionBeforeThrottle =>
```

Comment:
If capacity == 0 and feature is turned on and still no containers for action could be created, then turn on namespace throttling. If capacity == 0 and feature is turned off and no containers exist, just skip any considerations and turn on namespace throttling same as existing behavior

Reply:
I feel there is a subtle difference in the semantics when the feature is enabled.
Let's say there are one action utilizing 30 containers, and 15 actions with 1 container each.
The `totalContainers` is always bigger than 0. Then isn't throttling enabled?

The throttling would be enabled when an activation for another action arrives, and no container is created.
I think it would be better to enable the namespace throttling when the number of containers reaches `namespace limit * overprovisoinRatio`.

Reply:
the reasoning for this is so that activations can continue processing when the 15th action gets the last 1 container. If the 15th action can handle the throughput at this point it wouldn't want namespace throttling enabled so it can continue processing activations. You wouldn't want to stop activation processing until the 16th action comes in.

Or am I misremembering and namespace throttling only impacts new actions with 0 containers?

Reply:
> Or am I misremembering and namespace throttling only impacts new actions with 0 containers?

This is correct.
The namespace throttling is used when there is no action throttling data in ETCD which means no queue is created for the action.
https://github.com/apache/openwhisk/blob/master/core/controller/src/main/scala/org/apache/openwhisk/core/loadBalancer/FPCPoolBalancer.scala#L683

Reply:
I believe I have updated this behavior and added a commit to what you requested in the latest commit.

I think this pr should be good to go except for one remaining issue I just thought of if one of the 15 actions that got the overprovisioning slots and namespace throttling is enabled and one of those actions containers are removed then I'm not sure if namespace throttling will be correctly disabled, I just need to verify that before merging tomorrow.

Reply:
Okay added case to correctly disable namespace throttling if the container that frees up space is a container from the over-provisioning space in the latest commit. Would appreciate approval on this when you have some time @style95